File: code/community/1306409966936002566/debugging-403-error-gcp-bucket-signed-url.js
require('dotenv').config();
const fs = require('fs');
const { Deepgram } = require('@deepgram/sdk');

const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const deepgram = new Deepgram(deepgramApiKey);

const file = './audio-file.wav';

// Read audio file
const audio = fs.readFileSync(file);

// Transcribe audio
(async () => {
  try {
    const response = await deepgram.transcription.preRecorded(
      { buffer: audio, mimetype: 'audio/wav' },
      { language: 'en-US' }
    );
    console.log('Transcription:', response.results.channels[0].alternatives[0].transcript);
  } catch (error) {
    console.error('Error:', error);
  }
})();

File: code/community/1306409966936002566/debugging-403-error-gcp-bucket-signed-url.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Deepgram;
using Deepgram.Transcription;

class Program
{
    static async Task Main(string[] args)
    {
        string apiKey = Environment.GetEnvironmentVariable("DEEPGRAM_API_KEY");
        DeepgramClient deepgram = new DeepgramClient(apiKey);

        string path = "./audio-file.wav";

        // Read the audio file
        using (var audio = new FileStream(path, FileMode.Open))
        {
            // Transcribe the audio
            var response = await deepgram.Transcription.Prerecorded.TranscribeAudioAsync(
                audio,
                new Deepgram.Transcription.Prerecorded.TranscriptionOptions
                {
                    Language = "en-US"
                }
            );

            Console.WriteLine("Transcription: " + response.Results.Channels[0].Alternatives[0].Transcript);
        }
    }
}